Rework events page to include timeago

sinamics opened the pull request:
Resolves #5008
This pull request adds a timeago function that calculates the elapsed time since a given time string. The function currently only outputs the time elapsed in one unit of time (years, months, days, hours, minutes, seconds) based on the input time string. It does not take into account leap years or handle multiple units of time. 

Changes:
- Added timeago function to calculate elapsed time since a given time string
- Updated the function to correctly handle elapsed time in months
- New Clock icon

The event_end time is compared to the browser time
```
// Example 1:
const timeString1 = '2022-12-01T12:00:00Z';
// Output: "1 month ago"

// Example 2:
const timeString2 = '2022-01-01T12:00:00Z';
// Output: "1 year ago"

// Example 3:
const timeString3 = '2022-06-01T08:30:00Z';
// Output: "7 months ago"

// Example 4:
const timeString4 = '2022-01-14T23:56:07Z';
// Output: "2 hours ago"
```

Its not ideal to have all times in a oneliner, but it gives a more compact layout.
![Timeago](https://i.ibb.co/RDVGFNc/timeago.png)

sinamics committed:
new timeago component

sinamics committed:
added timeago to event

sinamics committed:
clock icon

sinamics committed:
clock icon. flex items center

NickM-27 commented:
Personally I am not a fan of putting it all together. On mobile this is going to turn into 3 lines of text and to me (personally) I don't care to know the time ago as the exact time is plenty.

I think this would be better implemented as a tooltip when hovering over the event or some other manner than a long string of text

NickM-27 reviewed (commented):
Personally I am not a fan of putting it all together. On mobile this is going to turn into 3 lines of text and to me (personally) I don't care to know the time ago as the exact time is plenty.

I think this would be better implemented as a tooltip when hovering over the event or some manner other than a long string of text

NickM-27 commented:
I think the hover should work, there's an example of that on the button component. 

I will say the clock icon is a nice touch

blakeblackshear commented:
We could shorten the minutes to m and the seconds to s as well.

sinamics committed:
dense prop

sinamics committed:
moved clipDuration. cleaner jsx, sm hidden

sinamics committed:
renamed clipduration => getDurationFromTimestamps

sinamics committed:
func description

sinamics commented:
Take two.

Changes:
- Moved the clipDuration to dateUtils and renamed to getDurationFromTimestamps. The events page is long as it is.
- New formatUnixTimestampToDateTime in dateUtils. gives better readability in Events page in my opinion.
- Changed duration "seconds, minutes, hours" => s, m, h
- Added dense option to TimeAgo component.
- "hidden md:inline" to both TimeAgo and Duration.  This will remove both on small devices. 

![take 2](https://i.ibb.co/fkgspVG/small.png)

I also tempted to test with a tooltip.. i might try this out later
anyway, what`s your thoughts?

NickM-27 commented:
That's great. My main thought is I liked the duration being inside `()` as it gives visual separation to the data as opposed to two `-`

But besides that I think this is coming together well.

sinamics commented:
yep, like this?
![layout](https://i.ibb.co/rGtcGrq/timeago.png)

NickM-27 commented:
That's what I was thinking, but if others disagree happy to hear other ideas

sinamics committed:
duration in parenthesis

sinamics commented:
it makes sense to put duration in parenthesis. I`ll add it for now.
Thx for the swift feedback.

mjalindahl commented:
Thanks @sinamics !  It looks great.
